src/test/java/org/openrewrite/staticanalysis/UseCollectionInterfacesTest.java

```diff
@@ -663,47 +664,51 @@ class Test {
     }
 
     @Test
-    void enumSet() {
+    @Issue("https://github.com/openrewrite/rewrite-static-analysis/issues/179")
+    void enumSetHasDifferentGenericTypeThanSet() {
         rewriteRun(
           //language=java
           java("public enum A {}"),
           //language=java
           java(
             """
               import java.util.EnumSet;
-              
+
               class Test {
                   public EnumSet values = EnumSet.allOf(A.class);
-              }
-              """,
-            """
-              import java.util.EnumSet;
-              import java.util.Set;
-              
-              class Test {
-                  public Set values = EnumSet.allOf(A.class);
+                  void iterate() {
+                      for (Enum<?> e : values) {
+                          // the for loop wouldn't compile anymore if the EnumSet declaration were changed
+                      }
+                  }
               }
               """
           )
         );
     }
 
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite-static-analysis/issues/179")
+    void enumSetNotInRecipe() {
+        assertThat(UseCollectionInterfaces.rspecRulesReplaceTypeMap).doesNotContainKey("java.util.EnumSet");
+    }
+
```

Comment:
Not sure we'd need this with the above test adapted already, right? Saves us from tying too strongly to implementation details (and having to maintain that) if we remove this one.
```suggestion
```